### PR TITLE
Add functionalities for scream ML correction

### PR DESCRIPTION
**Pull request by elynnwu**
Added a minimal set of ML correction functionality to `scream_run`, and `prognostic_scream_run` container is now capable of executing the unit test of scream calling python to predict ML correction. 

Added public API:
- `scream_run.steppers.machine_learning`: includes a minimal set of functionality to open a pre-trained ML model and getting a prediction form it

- [x] Tests added

Coverage reports (updated automatically):

### CODE REVIEW

File: workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    Attributes:
+        model: list of URLs to fv3fit models.
+        diagnostic_ml: do not apply ML tendencies if true.
+        scaling: if given, scale the outputs by the given factor. This is a manually
+            defined alteration of the model, and should not be used for
+            normalization.
+    Example::
+
+        MachineLearningConfig(
+            model=["gs://vcm-ml-scratch/test-annak/ml-pipeline-output"],
```
Comment (frodre) [inline workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py:34]: `models`

File: workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        self.models = models
+        if scaling is None:
+            self._scaling: Mapping[str, float] = {}
+        else:
+            self._scaling = scaling
+
+    @property
+    def input_variables(self) -> Set[str]:
+        vars = [model.input_variables for model in self.models]
+        return {var for model_vars in vars for var in model_vars}  # type: ignore
```
Comment (frodre) [inline workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py:65]:
```suggestion
        return {
            var 
            for model in self.models 
            for var in model.input_variables
        }  # type: ignore
```

Comment (frodre) [reply workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py:65]: Just trying to improve the clarity here since the return is hard to parse.   Another idea:

```
        all_inputs = []
        for model in self.models:
            all_inputs.extend(model.input_vars)
        return set(all_inputs)
```

File: workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+State = MutableMapping[Hashable, xr.DataArray]
+
+
+@dataclasses.dataclass
+class MachineLearningConfig:
+    """Machine learning configurations
+
+    Attributes:
+        model: list of URLs to fv3fit models.
```
Comment (frodre) [inline workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py:26]: Suggest using the plural to fit the description. `models`

File: workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        for model in self.models:
+            predictions.append(model.predict(arg))
+        ds = xr.merge(predictions)
+        for var, scale in self._scaling.items():
+            ds[var] *= scale
+        return ds
+
+
+def open_model(config: MachineLearningConfig) -> MultiModelAdapter:
+    model_paths = config.model
```
Comment (frodre) [inline workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py:78]: `config.models`

File: workflows/prognostic_scream_run/tests/test_ML_correction.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    dims = ["ncol", "z"]
+
+    data = xr.Dataset({"qv": (dims, arr), "T_mid": (dims, arr),})  # noqa: E231
+
+    return data
+
+
+def test_sample_scream_ml_config():
+    config = MachineLearningConfig(
+        model=["gs://vcm-ml-experiments/scream-n2f/test-scream-train-model"]
```
Comment (frodre) [inline workflows/prognostic_scream_run/tests/test_ML_correction.py:25]: `models`

File: workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    models = []
+    for path in model_paths:
+        model = cast(fv3fit.Predictor, fv3fit.load(path))
+        models.append(model)
+    return MultiModelAdapter(models, scaling=config.scaling)
+
+
+def download_model(config: MachineLearningConfig, path: str) -> Sequence[str]:
```
Comment (frodre) [inline workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py:86]: I know there are some temporary directory implementations to download and open models somewhere in fv3net rather than having to specify the path.  I don't know what context this function is used in, but consider if it's necessary to have the models in a specific place rather than removed upon completion of the run.

**Review by frodre — APPROVED** [review]
Thanks, @elynnwu.  LGTM.  Just a few minor comments and suggestions.

File: workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py (hunk 2 above)
Comment (elynnwu) [reply workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py:65]: I like this too. Definitely more clear this way.

File: workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py (hunk 6 above)
Comment (elynnwu) [reply workflows/prognostic_scream_run/scream_run/steppers/machine_learning.py:86]: Good point. This function is currently not used, and it's not clear yet how we are going to pass the pre-trained model to the actual scream run. For now, since this function is not tested, I will go ahead and remove. May add it back later if we end up needing to download the model first.
